### PR TITLE
Replace `request-debug` with `request-logger`

### LINKED ISSUE #12: Logging options override cached `request` object
When defining the logging options of `request` it modifies by reference the require cached object `request`, making all the system log requests that don't want to.

**Problem:** Any application that uses `request` and uses this client will log all the requests, even we do not want to log them. See [this](https://github.com/seegno/authy-client/blob/master/src/logging/request.js).

### PR DESCRIPTION
**Pull request by pedrobranco**
This PR fixes #12 and also removes the custom request serializer for`debugnyan`, and handles it directly before logging to `debugnyan`.

### CODE REVIEW

File: src/logging/request-obfuscator.js [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+
+const replacement = /(api_key=)([^&])*/;
+
+/**
+ * Export `RequestObfuscator`.
+ */
+
+export function obfuscate(request) {
+  // Obfuscate the API key on `uri`.
+  if (request.uri) {
```
Comment (pedrobranco) [inline src/logging/request-obfuscator.js]: Never.

File: src/logging/request-obfuscator.js [hunk 2]
```diff
@@ -0,0 +1,26 @@
+
+/**
+ * Module dependencies.
+ */
+
+import _ from 'lodash';
```
Comment (ruimarinho) [inline src/logging/request-obfuscator.js]: import { isString }

File: test/logging/request-obfuscator_test.js [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+
+      request.should.eql(_.defaults({ uri: 'foo=bar?api_key=*****' }, request));
+    });
+
+    it('should strip credentials from request `body`', () => {
+      const request = {
+        body: 'foo=bar?api_key=foobar',
+        id: '354f8341-eb27-4c91-a8f7-a30e303a0976',
+        method: 'GET',
+        uri: 'foo=bar?api_key=foobar'
```
Comment (ruimarinho) [inline test/logging/request-obfuscator_test.js]: Replace ? by &

File: test/logging/request-obfuscator_test.js [hunk 4]
```diff
@@ -0,0 +1,40 @@
+
+/**
+ * Module dependencies.
+ */
+
+import _ from 'lodash';
```
Comment (ruimarinho) [inline test/logging/request-obfuscator_test.js]: import { defaults } for consistency?

**Comment by ruimarinho** [comment]
Thank you for the PR. This looks awesome!

**Pull request merged**